src/main/scala/com/samstarling/prometheusfinagle/PrometheusStatsReceiver.scala

```diff
@@ -49,8 +49,8 @@ class PrometheusStatsReceiver(registry: CollectorRegistry,
 
   override def repr: AnyRef = this
 
-  override def counter(verbosity: Verbosity, name: String*): Counter = {
-    val (metricName, labels) = extractLabels(name)
+  override def counter(schema: CounterSchema): Counter = {
+    val (metricName, labels) = extractLabels(schema.metricBuilder.name)
```

Comment:
`schema.metricBuilder.name.toList` is safer because the pattern matching performed by the `extractLabels` method only works if you pass a `List` in. 

I have checked myself and the default finagle metrics are sometimes sent as an `ArrayBuffer` which makes the pattern matching fail and the`serviceName` ends up being concatenated to the final `metricName` instead of being handled as a separate label. 

The same applies to all different schemas so I'd change the `extractLabels` method to always convert a Seq to List instead.

Reply:
I don't think this issue was introduced by this patch because previously the varargs `name*` would always have been an Array.

IMO the best solution is to make the patterns work with any `Seq` (as appears to be the intent), rather than potentially waste time and ram allocating a `List`.  From a quick glance it looks like there is a stray use of `::` rather than `+:` in `DefaultMatch` but more investigation and testing is required and as I say I don't think it's relevant to this relevent to this particular PR.